### ISSUE TITLE
Installer Error Code 24

### ISSUE THREAD

**Issue by sebkolind** [TheRandomLabs/Scoop-Spotify#34]
I had the app running just fine, but then I restarted my machine, and all of a sudden when I try to open the app I get the `Installer Error Code: 24`. I saw that Spotify wanted to update the app (up in the right corner on the arrow icon) but I haven't touched that, since I figured it would break.

Now, after a restart I cannot open the app, since I am just getting this issue.

I have tried a `scoop update spotify-latest` without any success.

![image](https://user-images.githubusercontent.com/2152027/103872052-276ef400-50ce-11eb-9229-d3fad27cc731.png)

**Comment by TheRandomLabs** [comment]
Try reinstalling `spotify-latest` as admin:
`sudo scoop uninstall spotify-latest`
`sudo scoop install spotify-latest`

**Comment by sebastianks** [comment]
Thank you for getting back to me.

That did not seem to work. I get the same result, and if I click "Retry" it'll show me "Installing Spotify" for a very long time - I haven't actually seen it finish.

**Comment by TheRandomLabs** [comment]
Try doing the same, but with `--purge`: `sudo scoop uninstall --purge spotify-latest`.  This will remove all user data.

**Comment by sebastianks** [comment]
For now it looks like it is working, after I did the `uninstall` with `sudo`. I did try to uninstall without, and that did not work.

Thanks!

**Comment by sebastianks** [comment]
@TheRandomLabs Unfortunately, I do know have the same issue again. The installer opens and it does not move on. I am not sure why this is not working for me. Do you have any idea of what is going on?

**Comment by TheRandomLabs** [comment]
Hmm, not really. Did you install Spotify as administrator?

**Comment by sebastianks** [comment]
Yes, both `sudo` and in an elevated PS (I don't think that this makes a big difference though).

**Comment by TheRandomLabs** [comment]
Hmm, when you run `sudo scoop uninstall --purge spotify-latest`, can you confirm that `$env:APPDATA\Spotify` and `$env:LOCALAPPDATA\Spotify` are deleted?

**Comment by sebastianks** [comment]
Yes, they are deleted.

**Comment by TheRandomLabs** [comment]
I'm stumped.

**Comment by sebastianks** [comment]
Okay, fair. It seems like an issue on my machine - I might just install Spotify via the regular installer. Just triggers my OCD that I can't have everything installed via `scoop` you know? Anyway, thanks for your help 😉

**Comment by TheRandomLabs** [comment]
Yeah, I get that OCD. 😛 
You're welcome.

**Comment by TheRandomLabs** [comment]
If someone else has more information, let me know. 😉

**Comment by maxsubs** [comment]
I'm also getting this error. After solving the hash faile issue by reinstalling spotify-latest with --skip flag, I get this error, but not the first hours after installing it fresh. I get this like a day after I install spotify. When I get it the only things that I can do is:
```scoop uninstall spotify-latest --purge
scoop install spotify-latest```
and it works again for the next 24h (give or take)
What can it be? it wasn't happening before, should I try with an older version?

**Comment by TheRandomLabs** [comment]
Until I have a consistent way of reproducing this, I have no idea.

**Comment by maxsubs** [comment]
I don't know what's triggering it exactly, cause it always happens a few launch after I reinstall it. Is there any way to get some logs from launch? In the meantime I've schedule this dayly:
scoop uninstall spotify-latest --purge
scoop install spotify-latest --skip

I have these packages installed (in case some of them may be the problem):
spotify-latest blockthespot spicetify-cli genius-spicetify spicetify-themes

**Comment by maxsubs** [comment]
Since this error start appearing after the hash problem and we needed to use the --skip flag to get around it, couldn't these two things be related? I'd like to try to reinstalled spotify without the skip flag and see. Btw I've also tried the two older versions of spotify-latest + hold command, but didn't make any difference

**Comment by TheRandomLabs** [comment]
Hmm, maybe somehow Scoop is downloading an older version of Spotify, and Spotify is trying to update.

**Comment by dionysius** [comment]
I ran into this issue as well. According to the Readme:
> Spotify's built-in updater is disabled, and Scoop should be used to update it instead.

I've installed Spotify from this repo a big while ago. And I didn't use Spotify for a really good while (minimum a month, maybe minimum 2 months) on this machine until today. I don't use spicetify or any other package, only `spotify-latest`.

So, how I got into this issue: There was a **red dot** in Spotify asking me to install the update. This was odd, since I knew it was managed with scoop. So I closed Spotify and ran `scoop update *`, which also updated Spotify (more about this later). After opening again, I still got the red dot in Spotify and decided to fat finger that update. Spotify closed and now shows me the error like OP pictured, whenever I want to open it.

For reference my uninstallation and installation steps:
```
C:\Users\Dionysius
λ scoop uninstall spotify-latest
Uninstalling 'spotify-latest' (1.1.56.595.g2d2da0de).
Running uninstaller script...
Everyone: No mapping between account names and security IDs was done.
Removing shim for 'Spotify'.
Removing shortcut ~\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Scoop Apps\Spotify.lnk
Unlinking ~\scoop\apps\spotify-latest\current
Removing older version (1.1.48.625.g1c87c7f7).
Removing older version (1.1.53.608.g7ed9c03a).
Removing older version (1.1.55.498.gf9a83c60).
'spotify-latest' was uninstalled.

C:\Users\Dionysius
λ scoop install spotify-latest
Installing 'spotify-latest' (1.1.56.595.g2d2da0de) [64bit]
Loading SpotifyFullSetup.exe from cache
Checking hash of SpotifyFullSetup.exe ... ok.
Running installer script...
C:\Users\Dionysius\AppData\Local\Spotify\*: The system cannot find the path specified.
Everyone: No mapping between account names and security IDs was done.
Linking ~\scoop\apps\spotify-latest\current => ~\scoop\apps\spotify-latest\1.1.56.595.g2d2da0de
Creating shim for 'Spotify'.
Creating shortcut for Spotify (Spotify.exe)
Persisting config.ini
Running post-install script...
'spotify-latest' (1.1.56.595.g2d2da0de) was installed successfully!
Notes
-----
--purge should be used to fully uninstall this package.
```

1. Obervation: **My statement above about up-to-date spotify seems not to be true**. Unfortunately I don't have the output of the scoop update anymore, so I don't have any proof anymore about what exactly got updated. Maybe spotify update failed, but I didn't had an error.

So far spotify is fine again, I did not need to uninstall with `--purge` and my account login is still there.

So the question is maybe, what I've cited above: Was there maybe a version of spotify where the internal autoupdate was not disabled and on some machines it tried to automatically update? (Or maybe it would've happened to me as well, if I had closed and opened spotify again, but now we never know)

**Comment by orhanmaden** [comment]
In my case the scoop updater for spotify was not updated with latest version. I resolved temporary by removing the Spotify updater by navigating to `%localappdata%\Spotify\Update` and deleting the `exe` files.
